tests/rptest/services/redpanda.py

```diff
@@ -232,8 +232,21 @@ def start_redpanda(self, node):
 
         node.account.ssh(cmd)
 
-    def signal_redpanda(self, node, signal=signal.SIGKILL):
-        node.account.signal(self.redpanda_pid(node), signal, allow_fail=False)
+    def signal_redpanda(self, node, signal=signal.SIGKILL, idempotent=False):
+        """
+        :param idempotent: if true, then kill-like signals are ignored if
+                           the process is already gone.
+        """
+        pid = self.redpanda_pid(node)
+        if pid is None:
+            if idempotent and signal in {signal.SIGKILL, signal.SIGTERM}:
+                return
+            else:
+                raise RuntimeError(
+                    f"Can't signal redpanda on node {node.name}, it isn't running"
+                )
+
+        node.account.signal(pid, signal, allow_fail=False)
```

Comment:
it seems like this might also work.

```
allow_fail = idempotent and signal in {signal.SIGKILL, signal.SIGTERM}
node.account.signal(pid, signal, allow_fail)
```

`kill` will give a non-zero exit code if the process doesn't exist which is examined based on `allow_fail`

Reply:
I went with the more explicit pid=None check to print the friendliest message I could